### PR TITLE
Camp 71 as admin number of registered users

### PR DESCRIPTION
**Pull request by dabick14**
## Description
This PR implements a function and frontend pages for viewing the camp directory and the number of registrations per camp and level

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?
This has been tested on my local pc.

## Screenshots/Videos (if appropriate):

https://www.loom.com/share/d21c62cf7a3b4a4d863ffd80832398e1?sid=971b473b-4ba7-4ba5-8dfd-e3867d46da0e

### CODE REVIEW

File: functions/functions/src/aggregations.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import { getFirestore, FieldValue } from 'firebase-admin/firestore'
+import { Registration } from '../types'
+
+const db = getFirestore()
+
+const aggregateRegistration = async (registration: Registration) => {
+  const { campId, campusRef } = registration
```
Comment (ahadziii) [inline functions/functions/src/aggregations.ts]: Are we still using Ref instead of id

File: functions/functions/src/aggregations.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+            await planetReference.update({
+              registration: increment,
+              timestamp: FieldValue.serverTimestamp(),
+            })
+          }
+        }
+      }
+    }
+  } catch (error) {}
```
Comment (ahadziii) [inline functions/functions/src/aggregations.ts]: What happens after catching the error

Comment (ahadziii) [inline functions/functions/src/aggregations.ts]: What happens after catching the error

File: functions/functions/src/aggregations.ts (hunk 1 above)
Comment (dabick14) [reply functions/functions/src/aggregations.ts]: Changing this too. Thanks.

File: web-react-ts/src/components/RegisterUserCard.tsx [hunk 3]
```diff
@@ -29,36 +36,31 @@ const RegisterUserCard = ({ user }: { user: UserData }) => {
     currentUser?.email as string
   )
 
-  const { data: adminDoc } = useFirestoreDocData(adminDocReference)
+  const { data: adminDoc, status } = useFirestoreDocData(adminDocReference)
```
Comment (ahadziii) [inline web-react-ts/src/components/RegisterUserCard.tsx:39]: no type for status??

File: web-react-ts/src/pages/camps/AssignAdmin.tsx [hunk 4]
```diff
@@ -12,27 +9,20 @@ const AssignAdmin = () => {
...
-
-  const usersData: UserData[] = []
-
-  users?.forEach((user: DocumentData) => {
-    usersData.push({ id: user.id, ...user.data() })
-  })
+  const data = [
+    { id: '1', name: 'John Doe' },
+    { id: '2', name: 'Jane Doe' },
+  ]
```
Comment (ahadziii) [inline web-react-ts/src/pages/camps/AssignAdmin.tsx]: why dont you rather have a separate file to hold this JSON file , fetch from that file and maintain the previous code

File: web-react-ts/src/pages/directory/ContinentsByPlanet.tsx [hunk 5]
```diff
@@ -25,6 +32,8 @@ const ContinentsByEarth = () => {
     }
   )
 
+  console.log(continents)
```
Comment (ahadziii) [inline web-react-ts/src/pages/directory/ContinentsByPlanet.tsx]: Forgot to remove the log

File: web-react-ts/src/pages/directory/CountriesByContinents.tsx [hunk 6]
```diff
@@ -9,20 +9,24 @@ import {
...
+  const ref = doc(campRef, 'continents', continentId)
   const { data: continent } = useFirestoreDocData(ref)
 
-  const countriesCollection = collection(firestore, type)
+  const countriesCollection = collection(campRef, 'countries')
   const countriesQuery = query(
     countriesCollection,
-    where('continentRef', '==', continentId)
+    where('upperChurchId', '==', continentId)
     // orderBy('name', 'asc')
```
Comment (ahadziii) [inline web-react-ts/src/pages/directory/CountriesByContinents.tsx]: would you need the commented code?

File: web-react-ts/src/pages/directory/CountriesByContinents.tsx [hunk 7]
```diff
@@ -33,6 +37,8 @@ const CountriesByContinent = () => {
     }
   )
 
+  console.log('countries', countries)
+
```
Comment (ahadziii) [inline web-react-ts/src/pages/directory/CountriesByContinents.tsx]: forgot the logger

File: web-react-ts/src/utils/MenuDataTypes.ts [hunk 8]
```diff
@@ -6,7 +6,7 @@ interface DataItem {
 }
 
 interface ContinentsDataItem extends DataItem {
-  earthRef?: string
+  planetRef?: string
```
Comment (ahadziii) [inline web-react-ts/src/utils/MenuDataTypes.ts]: Change to id

File: web-react-ts/src/components/RegisterUserCard.tsx (hunk 3 above)
Comment (dabick14) [reply web-react-ts/src/components/RegisterUserCard.tsx:39]: @ahadziii  The adminDoc isn't a type, it's like a destructured object assignment.

File: functions/functions/src/directory-functions.ts [hunk 9]
```diff
@@ -24,173 +31,120 @@ const campusDirectory = async (camp: Camp, campId: string) => {
   await directoryRef.set(campusData)
 }
 
-const countryDirectory = async (camp: Camp, campId: string) => {
-  const countriesRef = db.collection('countries').doc(camp.levelId)
-  const countryDoc = await countriesRef.get()
+const planetDirectory = async (camp: Camp, campId: string) => {
+  const planetRef = db.collection('planets').doc(camp.levelId)
+  const planetDoc = await planetRef.get()
```
Comment (ahadziii) [inline functions/functions/src/directory-functions.ts:36]: planet ref?

File: functions/functions/src/directory-functions.ts [hunk 10]
```diff
@@ -24,173 +31,120 @@ const campusDirectory = async (camp: Camp, campId: string) => {
...
-    name: countryDoc.data()?.name,
-    countryRef: countryDoc.data()?.upperRef,
+  const planetData = {
+    name: planetDoc.data()?.name,
   }
 
-  // const campRef = db.collection('camps').doc(campId)
-  // await campRef.get()
+  const campRef = db.collection('camps').doc(campId)
+  await campRef.get()
```
Comment (ahadziii) [inline functions/functions/src/directory-functions.ts:43]: camp ref??

File: functions/functions/src/directory-functions.ts [hunk 11]
```diff
@@ -24,173 +31,120 @@ const campusDirectory = async (camp: Camp, campId: string) => {
...
-  // Create continents subcollection
-  const batch = db.batch()
-  const continentsDirectoryRef = campRef
-    .collection('continents')
-    .doc(continentDocId)
-  batch.set(continentsDirectoryRef, {
-    ...continentData,
-    type: 'continent', // add a type field to identify the entity type
-  })
+  const continentsRef = db.collection('continents')
```
Comment (ahadziii) [inline functions/functions/src/directory-functions.ts:53]: Continent ref?

File: web-react-ts/src/components/RegisterUserCard.tsx [hunk 12]
```diff
@@ -29,36 +36,31 @@ const RegisterUserCard = ({ user }: { user: UserData }) => {
...
 
   const adminCamp = adminDoc?.camp_admin.find(
     (camp: UserCampData) =>
       camp.campId === campId && camp.role?.includes('campusAdmin')
   )
 
+  // console.log('adminCamp', adminCamp)
```
Comment (ahadziii) [inline web-react-ts/src/components/RegisterUserCard.tsx]: remove unused code

**Review by ahadziii — APPROVED** [review]

**Pull request merged**